src/components/products/list/ProductsList.js

```diff
@@ -10,155 +10,115 @@ import ProductGrid from '../components/grid/ProductGrid'
 import LoadingView from '../../common/LoadingView'
 import './ProductsList.scss'
 
-class ProductsList extends Component {
-	constructor(props) {
-		super(props)
-
-		this.state = {
-			products: [],
-			isProductCreateModalOpen: false,
-			renderListView: true,
-			isLoaded: false,
-		}
-
-		this.refreshList = this.refreshList.bind(this)
-		this.openProductModal = this.openProductModal.bind(this)
-		this.closeProductModal = this.closeProductModal.bind(this)
-		this.renderListView = this.renderListView.bind(this)
-		this.renderGridView = this.renderGridView.bind(this)
-	}
-
-	async componentDidMount() {
-		const prods = await this.getAllProducts()
-		this.setState({ products: prods, isLoaded: true })
-	}
+function ProductsList(props) {
+	const [isLoaded, setIsLoaded] = useState(false)
 
-	async getAllProducts() {
-		const response = await ProductsApiService.getAllProducts()
-		return this.getProductsReactiveObjectsList(response.data)
-	}
+	const [products, setProducts] = useState([])
+	const [listView, setListView] = useState(true)
+	const [isProductCreateModalOpen, setIsProductCreateModalOpen] = useState(false)
 
-	getProductsReactiveObjectsList(productsList) {
+	const getProductsReactiveObjectsList = productsList => {
 		return productsList.map(product => {
 			const productRO = { ...product }
 
-			productRO.editHandler = e => {
-				e.stopPropagation()
-				const editUrl = `products/${product.id}/edit`
-				this.props.history.push(editUrl)
+			productRO.clickHandler = () => {
+				props.history.push(`products/${product.id}`)
 			}
 
-			productRO.clickHandler = () => {
-				this.props.history.push(`products/${product.id}`)
+			productRO.editHandler = event => {
+				event.stopPropagation()
+				const editUrl = `products/${product.id}/edit`
+				props.history.push(editUrl)
 			}
 
 			productRO.deleteHandler = async productId => {
 				const deleteResult = await ProductsApiService.deleteProduct(productId)
 
 				if (deleteResult !== undefined && deleteResult.status === 200) {
-					this.refreshList()
+					const fetchedProducts = await ProductsApiService.getAllProducts()
```

Comment:
this code also repeats twice, that's why there was refreshList() before, could you bring it back so we don't repeat the code